model.py

```diff
@@ -166,6 +168,64 @@ class SoftplusLayer(tf.keras.layers.Layer):
     def call(self, inputs):
         return tf.math.log(1 + tf.exp(inputs))
 
+def cross_val_with_best_model(
+    model_fn: Callable,
+    sp: ScaledPartitions,
+    n_cv_folds: int,
+    epochs_per_fold: int,
+    batch_size: int,
```

Comment:
Perhaps we could add a `fit_kwargs` field here so that we don't need DNN/Keras specifics like `batch_size` and `callbacks_list`, but still can support direct training on these APIs without adding a wrapper class.

Then, we would invoke `model.fit(X_train, Y_train, validation_data=(X_val, Y_val,), epochs=epochs_per_fold, **fit_kwargs)`